src/Math/babylon.math.ts

```diff
@@ -7187,7 +7187,7 @@
         public static Color4: Color4[] = [new Color4(0, 0, 0, 0), new Color4(0, 0, 0, 0)];
         public static Vector2: Vector2[] = [Vector2.Zero(), Vector2.Zero(), Vector2.Zero()];  // 3 temp Vector2 at once should be enough
         public static Vector3: Vector3[] = [Vector3.Zero(), Vector3.Zero(), Vector3.Zero(),
-        Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero()];    // 9 temp Vector3 at once should be enough
+        Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero()]; // 11 temp Vector3 at once should be enough

```

Comment:
when typescript is upgraded to version 3.0+, you should consider adding someting like :
````
const buildArray = <T, P extends any[]>(size:number, konstructor: new (...params: P)=>T, ...params: P):Array<T> => {
    const a = new Array<T>(size);
    for (let i = 0; i < size; ++i)
        a[i] = new konstructor(...params);
    return a;
}
````

so that instead of writing
````
public static Vector3: Vector3[] = [Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero(), Vector3.Zero()];
````
you could write
````
public static Vector3: Vector3[] = buildArray(11, Vector3);
````

Reply:
yep!!